WordPress/src/main/java/org/wordpress/android/ui/deeplinks/DeepLinkingIntentReceiverActivity.java

```diff
@@ -49,6 +50,17 @@ public class DeepLinkingIntentReceiverActivity extends LocaleAwareActivity {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
+        OnBackPressedCallback callback = new OnBackPressedCallback(true) {
+            @Override
+            public void handleOnBackPressed() {
+                setEnabled(false);
+                getOnBackPressedDispatcher().onBackPressed();
+                finish();
```

Comment:
Question (❓): I was trying to figure out what's the best way to deal with both, replacing these `super.onBackPressed()` calls and the `finish()` calls, thus I have some question about both, see below:

1. `super.onBackPressed()`: Is `setEnabled(false)` + `getOnBackPressedDispatcher().onBackPressed()` the recommended way to migrate those calls when utilizing `OnBackPressedCallback`. Do you maybe have some references on that so I could take a quick look? Unfortunately, I couldn't find any references on using this pattern myself. PS: No matter, from my quick testing, this indeed seems to be doing the trick. I am just not sure if that't the best we can do atm, especially after taking a look at [this](https://www.droidcon.com/2023/02/22/handling-back-press-in-android-13-the-correct-way/).
2. `finish()`: After me taking a look at [this](https://www.droidcon.com/2023/02/22/handling-back-press-in-android-13-the-correct-way/), the author mentioned that adding such calls will break the predictive back gesture feature. I know, we don't use `android:enableOnBackInvokedCallback="true"` atm, but that made me wonder if we can do better anyway, and avoid redoing that later on, wdyt? 🤔 

FYI: About no.2 above, in addition to this class and after me doing a quick search, I found the below additional activities to be using such `finish()` call:
- `JetpackSecuritySettingsActivity` via `finish()`
- `PostSettingsTagsActivity` via `saveAndFinish()`
- `SelectCategoriesActivity` via `saveAndFinish()`
- `JetpackConnectionResultActivity` via `finishAndGoBackToSource()`

Reply:
Thank you for initiating this discussion. I have spent a lot of time thinking about how to migrate `super.onBackPressed()`. Unfortunately, I couldn't find any official recommendations for migrating `super.onBackPressed()`. However, did come across some [references](https://issuetracker.google.com/issues/247982487) that do not recommend the current solution here if the predictive gesture feature is enabled.
I believe that the solution proposed here simply mimics the behavior of `super.onBackPressed()`, but it will have side effects when `enableOnBackInvokedCallback` is set to `true`. Therefore, it appears safe to me, but I would appreciate a review of my thought process.

---
**The problem**
When the back button is pressed,
- If an enabled `OnBackPressedCallback` exists, the system does nothing. It does not finish the activity, but only calls the callback function.
- If no enabled `OnBackPressedCallback` exists, the system handles the back button and finishes the activity.

We want the system to handle the back button, but we also want to run our custom code before the system. 

**The current solution**
I looked for a more elegant solution, but unfortunately, I could not find one. Therefore, I disabled the `OnBackPressedCallback` (`setEnabled(false)`) and then called `getOnBackPressedDispatcher().onBackPressed()` again to allow the system to handle the back button and keep the behavior of `super.onBackPressed()`. 

**A small concern for the current solution**
My only concern is that we may still need the callback after calling the `getOnBackPressedDispatcher().onBackPressed()`, but the callback will be disabled in this case. If `getOnBackPressedDispatcher().onBackPressed()` finishes the activity, there will be no issue. However, the system may just [collapse the drawer](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/app/Activity.java#3958) with the back button. I have checked our cases and could not find any case where this issue occurred. We can still add `setEnabled(true)` after `getOnBackPressedDispatcher().onBackPressed()` just in case, but I am of the opinion that it is not necessary. **Wdyt?**.

---

>  adding such calls will break the predictive back gesture feature. I know, we don't use android:enableOnBackInvokedCallback="true" atm, but that made me wonder if we can do better anyway, and avoid redoing that later on, wdyt? 🤔

`finish()` is breaking the predictive back feature, and `setEnabled(false) + getOnBackPressedDispatcher().onBackPressed()` is also [breaking](https://issuetracker.google.com/issues/247982487#comment3) it. Therefore, if I don't miss anything, there's no way to use custom back navigation and finish your activity. 😅
I'm aware of the risk of redoing some work here. However,  I prefer to keep the `predictive back gesture` out of [updating compileSdk issue](https://github.com/wordpress-mobile/WordPress-Android/pull/17947)'s scope. My other reasons for not enabling it are that the predictive gesture feature also needs upgrading to `androidx.activity:activity:1.6.0`, and it is a feature for Android 14. We can consider predictive back gesture as a separate project. 

---
**Summary:** I prefer to keep the current implementation if you cannot find any side effects of it like me.

Reply:
👋 @irfano !

> Thank you for initiating this discussion. I have spent a lot of time thinking about how to migrate super.onBackPressed(). Unfortunately, I couldn't find any official recommendations for migrating super.onBackPressed(). However, did come across some [references](https://issuetracker.google.com/issues/247982487) that do not recommend the current solution here if the predictive gesture feature is enabled.

Great find on this [references](https://issuetracker.google.com/issues/247982487) that also don't recommend the current solution. 👍 

> I believe that the solution proposed here simply mimics the behavior of super.onBackPressed(), but it will have side effects when enableOnBackInvokedCallback is set to true. Therefore, it appears safe to me, but I would appreciate a review of my thought process.

It indeed seems safe until we enable the predictive back gesture, that was my only concern. 👍 

> The problem

👍 

> The current solution

👍 

> A small concern for the current solution

I have the exact same concern, but I was also on the fence about adding this `setEnabled(true)` after `getOnBackPressedDispatcher().onBackPressed()`. But maybe we should, just to be on the safe side, not sure... 🤔

Suggestion (💡): An additional thought I had on that, just because of the fact that we keep replicating this behavior in multiple places, what if we create a few extension functions, like the below `AppCompatActivity` related one, and use them instead:

`AppCompatActivity`
```
/**
 * This is a temporary workaround for the issue described here: https://issuetracker.google.com/issues/247982487
 *
 * TODO: Replace this temporary workaround before enabling predictive back gesture on this project
 *  (android:enableOnBackInvokedCallback="true").
 *
 * Related Issue: <LINK>
 */
fun AppCompatActivity.onBackPressedCompat(onBackPressedCallback: OnBackPressedCallback) {
    onBackPressedCallback.isEnabled = false
    onBackPressedDispatcher.onBackPressed()
    onBackPressedCallback.isEnabled = true
}
```

> finish() is breaking the predictive back feature, and setEnabled(false) + getOnBackPressedDispatcher().onBackPressed() is also [breaking](https://issuetracker.google.com/issues/247982487#comment3) it. Therefore, if I don't miss anything, there's no way to use custom back navigation and finish your activity. 😅

😅 Yea, I too am not sure about it atm, I would have to research that a bit more to figure out if there is a way to use a custom back navigation and finish your activity. 🤷 

> I'm aware of the risk of redoing some work here. However, I prefer to keep the predictive back gesture out of https://github.com/wordpress-mobile/WordPress-Android/pull/17947's scope. My other reasons for not enabling it are that the predictive gesture feature also needs upgrading to androidx.activity:activity:1.6.0, and it is a feature for Android 14. We can consider predictive back gesture as a separate project.

For sure, my intention with this comment of mine wasn't about us enabling predictive back gesture here, not at all, I would actually advice against that, this is a whole other work that we would only be able to do in the future (as you noted too). 💯 

However, now that we are at it, I would advise us to create that future predictive back gesture issue, noting all these details and reference this issue on the KDoc comments that we might add as part of the extension functions idea that I suggested above. Wdyt?

> Summary: I prefer to keep the current implementation if you cannot find any side effects of it like me.

👍

Reply:
> Suggestion (💡): An additional thought I had on that, just because of the fact that we keep replicating this behavior in multiple places, what if we create a few extension functions

I was just about to do that before reading your comment. It's good that we're thinking alike. Done: bfcf01d2e320bba9894ace5677b6d43b865584f5

I opened the [issue](https://github.com/wordpress-mobile/WordPress-Android/issues/18053) as you suggested. I also received a [response](https://stackoverflow.com/questions/75462234/proper-usage-of-onbackpresseddispatcher/75462987#comment133451000_75462987) from a Google Engineer. According to the answer, we should update all of our back-pressed approaches. Since the purpose of custom back navigation differs for each class, the solution should also be different. Still, I'm for leaving that change to the other [issue](https://github.com/wordpress-mobile/WordPress-Android/issues/18053).

Reply:
This is awesome @irfano and thank you so much for going through this change, it does feel better indeed! 🙇  ❤️ 🚀